organisations/schema.py

```diff
@@ -36,35 +36,76 @@ class Meta:
         interfaces = (relay.Node,)
 
 
+def validate_person_data(kwargs):
+    # TODO: Add validation
+    pass
+
+
+class CreateMyProfileMutation(graphene.relay.ClientIDMutation):
+    class Input:
+        name = graphene.String(required=True)
+        phone_number = graphene.String()
+        email_address = graphene.String(required=True)
+        organisations = graphene.List(graphene.ID)
+
+    my_profile = graphene.Field(PersonNode)
+
+    @classmethod
+    @login_required
+    @transaction.atomic
+    def mutate_and_get_payload(cls, root, info, **kwargs):
+        user = info.context.user
+        if Person.objects.filter(user=user).exists():
+            raise ApiUsageError("User profile already exists")
+        validate_person_data(kwargs)
+        organisation_ids = kwargs.pop("organisations", None)
+        kwargs["user_id"] = user.id
+        person = Person.objects.create(**kwargs)
+        if organisation_ids:
+            for org_global_id in organisation_ids:
+                org_id = get_node_id_from_global_id(org_global_id, "OrganisationNode")
+                try:
+                    organisation = Organisation.objects.get(id=org_id)
+                except Organisation.DoesNotExist as e:
+                    raise ObjectDoesNotExistError(e)
+                person.organisations.add(organisation)
+        return CreateMyProfileMutation(my_profile=person)
+
+
 class UpdateMyProfileMutation(graphene.relay.ClientIDMutation):
     class Input:
         name = graphene.String()
         phone_number = graphene.String()
         email_address = graphene.String()
+        organisations = graphene.List(graphene.ID)
 
     my_profile = graphene.Field(PersonNode)
 
     @classmethod
     @login_required
     @transaction.atomic
     def mutate_and_get_payload(cls, root, info, **kwargs):
+        validate_person_data(kwargs)
         user = info.context.user
-
+        organisation_ids = kwargs.pop("organisations", None)
         try:
             person = user.person
         except Person.DoesNotExist as e:
             raise ObjectDoesNotExistError(e)
 
         update_object(person, kwargs)
+        if organisation_ids:
+            for org_global_id in organisation_ids:
+                org_id = get_node_id_from_global_id(org_global_id, "OrganisationNode")
+                try:
+                    organisation = Organisation.objects.get(id=org_id)
+                except Organisation.DoesNotExist as e:
+                    raise ObjectDoesNotExistError(e)
+                person.organisations.add(organisation)
```

Comment:
Hmm, now it doesn't seem to be possible to remove organisations the person already has, just to add new ones. Is that the desired functionality?

Reply:
No, it's not. Organisations should be cleared and re-added if necessary. Nice catch 👍